library/src/androidTest/java/org/xmtp/android/library/ConversationsTest.kt

```diff
@@ -228,10 +228,76 @@ class ConversationsTest {
                 )
             )
             assertEquals(
-                alixClient2.preferences.consentList.conversationState(dm2.id),
+                alixClient2.preferences.conversationState(dm2.id),
                 ConsentState.ALLOWED
             )
             assertEquals(dm2.consentState(), ConsentState.ALLOWED)
         }
     }
+
+    @Test
+    fun testStreamConsent() {
+        val key = SecureRandom().generateSeed(32)
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val alixWallet = PrivateKeyBuilder()
+
+        val alixClient = runBlocking {
+            Client().create(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key
+                )
+            )
+        }
+        val alixGroup = runBlocking { alixClient.conversations.newGroup(listOf(bo.walletAddress)) }
+
+        val alixClient2 = runBlocking {
+            Client().create(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key,
+                    dbDirectory = context.filesDir.absolutePath.toString()
+                )
+            )
+        }
+
+        runBlocking {
+            alixGroup.send("Hello")
+            alixClient2.conversations.sync()
+            alixClient.conversations.syncAllConversations()
+            alixClient2.conversations.syncAllConversations()
+        }
+        val alix2Group = alixClient2.findGroup(alixGroup.id)!!
+        val consent = mutableListOf<ConsentListEntry>()
+        val job = CoroutineScope(Dispatchers.IO).launch {
+            try {
+                alixClient.preferences.streamConsent()
+                    .collect { entry ->
+                        consent.add(entry)
+                    }
+            } catch (e: Exception) {
+            }
+        }
+
+        Thread.sleep(1000)
+
+        runBlocking {
+            alix2Group.updateConsentState(ConsentState.DENIED)
+            val dm3 = alixClient2.conversations.newConversation(caro.walletAddress)
+            dm3.updateConsentState(ConsentState.DENIED)
+            alixClient.conversations.sync()
+            alixClient2.conversations.sync()
+            alixClient.conversations.syncAllConversations()
+            alixClient2.conversations.syncAllConversations()
+        }
+
+        Thread.sleep(2000)
+        assertEquals(3, consent.size)
```

Comment:
Checking if I understand - there are 3 consent entries because when `alixClient2` creates the `dm3` group with caro, it automatically makes an entry of ALLOWED, and the other two entries are making the `alix2Group` DENIED, and setting the `dm3` group DENIED. Is that right?

Reply:
Correct!